provider/src/main/scala/com/actian/spark_vector/provider/RequestHandler.scala

```diff
@@ -92,17 +93,21 @@ class RequestHandler(sqlContext: SQLContext, val auth: ProviderAuth) extends Log
     writeJobResult(result)
   }
 
+  private def getMessage(cause: Throwable): String = cause.getMessage + (if (cause.getCause != null) s", caused by: ${getMessage(cause.getCause)}" else "")
+
   /** Handle the failure of a query request */
   private def handleFailure(cause: Throwable)(implicit socket: SocketChannel) = closeResourceAfterUse(socket) {
     val result = cause match {
       case JobException(e, job, part) => {
-        logInfo(s"Job tr_id:${job.transaction_id}, query_id:${job.query_id} failed for part ${part.part_id}. Reason: ${cause.getMessage}")
+        val message = getMessage(e)
+        logInfo(s"Job tr_id:${job.transaction_id}, query_id:${job.query_id} failed for part ${part.part_id}. Reason: $message")
```

Comment:
Decide whether to use logInfo (line 103) or logDebug (line 104) because both have similar info to print

Also, either logInfo or logDebug accepts a Throwable obj as a 2nd param; `logInfo(string, t)` works too. I don't think we need `Reason: ${message}` there since `message` is fully sent throughout JobResult, unless we really want to print the entire cause chain here too. If so, why down below we logError(...) with cause and not getMessage(cause)?